crates/database/src/fee_policies.rs

```diff
@@ -0,0 +1,175 @@
+use {
+    crate::{auction::AuctionId, OrderUid, PgTransaction},
+    sqlx::PgConnection,
+    std::ops::DerefMut,
+};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FeePolicy {
+    pub auction_id: AuctionId,
+    pub order_uid: OrderUid,
+    pub kind: FeePolicyKind,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum FeePolicyKind {
+    PriceImprovement {
+        price_improvement_factor: f64,
+        max_volume_factor: f64,
+    },
+    Volume {
+        factor: f64,
+    },
+}
+
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+struct FeePolicyRow {
+    auction_id: AuctionId,
+    order_uid: OrderUid,
+    kind: FeePolicyKindRow,
+    price_improvement_factor: Option<f64>,
+    max_volume_factor: Option<f64>,
+    volume_factor: Option<f64>,
+}
+
+#[derive(Debug, Clone, PartialEq, sqlx::Type)]
+#[sqlx(type_name = "PolicyKind", rename_all = "lowercase")]
+enum FeePolicyKindRow {
+    PriceImprovement,
+    Volume,
+}
+
+pub async fn insert(ex: &mut PgTransaction<'_>, fee_policy: FeePolicy) -> Result<(), sqlx::Error> {
+    const QUERY: &str = r#"
+        INSERT INTO fee_policies (auction_id, order_uid, kind, price_improvement_factor, max_volume_factor, volume_factor)
+        VALUES ($1, $2, $3, $4, $5, $6)
+    "#;
+    let fee_policy = FeePolicyRow::from(fee_policy);
+    sqlx::query(QUERY)
+        .bind(fee_policy.auction_id)
+        .bind(fee_policy.order_uid)
+        .bind(fee_policy.kind)
+        .bind(fee_policy.price_improvement_factor)
+        .bind(fee_policy.max_volume_factor)
+        .bind(fee_policy.volume_factor)
+        .execute(ex.deref_mut())
+        .await?;
+    Ok(())
+}
+
+pub async fn fetch(
+    ex: &mut PgConnection,
+    auction_id: AuctionId,
+    order_uid: OrderUid,
+) -> Result<Vec<FeePolicy>, sqlx::Error> {
+    const QUERY: &str = r#"
+        SELECT * FROM fee_policies
+        WHERE auction_id = $1 AND order_uid = $2
+        ORDER BY application_order
+    "#;
+    let rows = sqlx::query_as::<_, FeePolicyRow>(QUERY)
+        .bind(auction_id)
+        .bind(order_uid)
+        .fetch_all(ex)
+        .await?
+        .into_iter()
+        .map(Into::into)
+        .collect();
+    Ok(rows)
+}
+
+impl From<FeePolicyRow> for FeePolicy {
```

Comment:
similar to other comment, adjust this to convert to domain fee policy object and move it to autopilot infra.